### ISSUE TITLE
Use same seed for random search when comparing pipelines and only tuning the machine learning algorithm

### ISSUE THREAD

**Issue by mfeurer** [slds-lmu/lecture_appml#45]
Currently, the task 03_hw1_impact_encoding_py compares two feature preprocessing methods and tunes XGBoost for them. However, different seeds are used for the random search, leading to different hyperparameters of XGBoost. This makes it unclear if one of them is actually better than the other.
